controller/sync.go

```diff
@@ -236,6 +236,8 @@ func (m *appStateManager) SyncAppState(app *v1alpha1.Application, state *v1alpha
 		sync.WithResourceModificationChecker(syncOp.SyncOptions.HasOption("ApplyOutOfSyncOnly=true"), compareResult.diffResultList),
 		sync.WithPrunePropagationPolicy(&prunePropagationPolicy),
 		sync.WithReplace(syncOp.SyncOptions.HasOption(common.SyncOptionReplace)),
+		sync.WithServerSideApply(syncOp.SyncOptions.HasOption(common.SyncOptionServerSideApply)),
```

Comment:
Since server side apply is another sync option, does it need to be added to the UI as well? 
If we need it, do we want to add it in this pr or another one?

![image](https://user-images.githubusercontent.com/25150124/177223870-2fcc2154-d0e3-4230-8551-4c9d66a8dd49.png)

Reply:
added the option